dbt/adapters/bigquery.py

```diff
@@ -0,0 +1,300 @@
+    @classmethod
+    def initialize(cls):
+        google = cls._import('google')
+        google.auth = cls._import('google.auth')
+        google.oauth2 = cls._import('google.oauth2')
+
+        google.cloud = cls._import('google.cloud')
+        google.cloud.bigquery = cls._import('google.cloud.bigquery')
+        google.cloud.exceptions = cls._import('google.cloud.exceptions')
+
+        globals()['google'] = google
+
+    @classmethod
+    def get_materializer(cls, node, existing):
+        materializer = dbt.materializers.NonDDLMaterializer
+        return dbt.materializers.make_materializer(materializer,
+                                                   cls,
+                                                   node,
+                                                   existing)
+
+    @classmethod
+    def handle_error(cls, error, message, sql):
+        logger.debug(message.format(sql=sql))
+        logger.debug(error)
+        error_msg = "\n".join([error['message'] for error in error.errors])
+        raise dbt.exceptions.RuntimeException(error_msg)
+
+    @classmethod
+    @contextmanager
+    def exception_handler(cls, profile, sql, model_name=None,
+                          connection_name='master'):
+        try:
+            yield
+
+        except google.cloud.exceptions.BadRequest as e:
+            message = "Bad request while running:\n{sql}"
+            cls.handle_error(e, message, sql)
+
+        except google.cloud.exceptions.Forbidden as e:
+            message = "Access denied while running:\n{sql}"
+            cls.handle_error(e, message, sql)
+
+        except Exception as e:
+            logger.debug("Unhandled error while running:\n{}".format(sql))
+            logger.debug(e)
+            raise dbt.exceptions.RuntimeException(e)
+
+    @classmethod
+    def type(cls):
+        return 'bigquery'
+
+    @classmethod
+    def date_function(cls):
+        return 'CURRENT_TIMESTAMP()'
+
+    @classmethod
+    def begin(cls, profile, name='master'):
+        pass
+
+    @classmethod
+    def commit(cls, connection):
+        pass
+
+    @classmethod
+    def get_status(cls, cursor):
+        raise Exception("Not implemented")
+        state = cursor.sqlstate
+
+        if state is None:
+            state = 'SUCCESS'
+
+        return "{} {}".format(state, cursor.rowcount)
+
+    @classmethod
+    def get_bigquery_credentials(cls, config):
+        method = config.get('method')
+        Creds = google.oauth2.service_account.Credentials
+
+        if method == 'oauth':
+            return None
+
+        elif method == 'service-account':
+            keyfile = config.get('keyfile')
+            return Creds.from_service_account_file(keyfile)
+
+        elif method == 'service-account-json':
+            details = config.get('config')
+            return Creds.from_service_account_info(details)
+
+        error = ('Bad `method` in profile: "{}". '
+                 'Should be "oauth" or "service-account"'.format(method))
```

Comment:
also can we put this validation into a voluptuous schema for the bigquery profile?